plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestParquetPageSkipping.java

```diff
@@ -89,6 +89,11 @@ public void testAndPredicates()
         buildSortedTables(tableName, "totalprice", "double");
         int rowCount = assertSameResults("SELECT * FROM " + tableName + " WHERE totalprice BETWEEN 100000 AND 131280 AND clerk = 'Clerk#000000624'");
         assertThat(rowCount).isGreaterThan(0);
+
+        // `totalprice BETWEEN 51890 AND 51900` is chosen to lie between min/max values of row group
+        // but outside page level min/max boundaries to trigger pruning of row group using column index
```

Comment:
is it verifiable?
is it always true? ie can we write data slightly differently, or change writer defaults?

Reply:
For verification I think we'd have to look at bytes read from operator stats, I avoided that because we know tests based on that to be flaky.
Regarding changes to writer config, page indexes aren't being written in new writer (https://github.com/trinodb/trino/issues/9359), but I checked that changing `compression-codec` wouldn't affect the test. Also, buildSortedTables hard codes writer configs so we shouldn't be affected by change in default values of existing configs.
I think all the existing tests in this class rely on data to be written in certain layout and skip verification of whether filtering actually took place. 
Maybe we can improve this to include verification of filtering based on operator stats after those become reliable for tests ?

Reply:
> For verification I think we'd have to look at bytes read from operator stats, I avoided that because we know tests based on that to be flaky.

that's good point. see https://github.com/trinodb/trino/pull/9338#discussion_r714872335

Reply:
I've added assertions now to verify that row group pruning took place